segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -1066,25 +1065,26 @@ public void readTableEntriesDelta(WireCommands.ReadTableEntriesDelta readTableEn
 
     }
 
-    private void truncateIndexSegment(String segment, long offset) {
+    private CompletableFuture<Void> truncateIndexSegment(String segment, long offset) {
         String indexSegment = getIndexSegmentName(segment);
+        long indexSegmentOffset;
         try {
-            long indexSegmentOffset = IndexRequestProcessor.locateOffsetForIndexSegment(segmentStore, segment, offset, false);
-            log.debug("Truncating index segment {} at offset {}.", indexSegment, indexSegmentOffset);
-            if (indexSegmentOffset == 0) {
-                return;
-            }
-            segmentStore.truncateStreamSegment(indexSegment, indexSegmentOffset, TIMEOUT)
-                    .whenComplete((v, exp) -> {
-                        if (exp != null) {
-                            log.warn("Failed to truncate index segment {} at offset {} due to", indexSegment, indexSegmentOffset, exp);
-                        } else {
-                            log.info("Successfully truncated index segment {} at offset {}", indexSegment, indexSegmentOffset);
-                        }
-                    });
+            indexSegmentOffset = IndexRequestProcessor.locateOffsetForIndexSegment(segmentStore, segment, offset, false);
+            log.info("Truncating index segment {} at offset {}.", indexSegment, indexSegmentOffset);
         } catch (Exception e) {
-            log.warn("Unable to locate offset for index segment {}  for offset {} due to ", indexSegment, offset, e);
+            Throwable ex = Exceptions.unwrap(e);
+            if (ex instanceof StreamSegmentNotExistsException) {
+                log.info("Stream segment {} does not exists and cannot perform truncate index segment operation", segment);
```

Comment:
Can you please update log message "log.info("Stream segment {} does not exists, so skipping truncation of index segment.", segment);"